### PR TITLE
Allow to disable SSL certificate verification in CLI

### LINKED ISSUE #18: Allow to disable SSL verification during upload
Behind a corporate firewall with self-signed SSL certificates, upload fails with the following issue:

```
requests.exceptions.SSLError: HTTPSConnectionPool(host='my-conda-host.com', port=443): Max retries exceeded with url: /api/channels/my-channel/files/ (Caused by SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: self-signed certificate in certificate chain (_ssl.c:992)')))
```

Would be nice to have a flag to disable SSL verification. This cannot be done via environment variable anymore since `requests>=2.28.0` (see https://github.com/psf/requests/pull/6074).

### PR DESCRIPTION
**Pull request by borchero**
Resolves #18

### CODE REVIEW

**Review by janjagusch — CHANGES_REQUESTED** [review]
Thanks for your PR, @borchero. I prefer not to include this option to the `post_file_to_channel` method. It would be surprising for users to see that only a single method supports that option. I also don't want to actively encourage to disable SSL verification.

I you need to disable SSL verification, I suggest the following:

```py
client = QuetzClient.from_token(...)
client.session.verify = False
```

**Comment by borchero** [comment]
@janjagusch my main goal was to have this functionality in the CLI: any idea how I can achieve that instead? It's rather ugly to create my own "wrapper CLI"...

In general, I wouldn't expect many people to use quetz-client as a Python package (?) 🤔

> I also don't want to actively encourage to disable SSL verification.

I also don't think that's what's happening here :smile: most tools (including `mamba`) have such a flag since you, unfortunately, need it sometimes.

> It would be surprising for users to see that only a single method supports that option.

I think we could easily make this a global config option.

**Comment by janjagusch** [comment]
It should be simple to add this feature to the CLI. You only have to add a `verify` argument here:
https://github.com/mamba-org/quetz-client/blob/00a0a7ccc22f72f0ec8fea8a6e0f977f84fa2157/src/quetz_client/cli.py#L8

That's something I would see in scope for this package and would appreciate a contribution for. :)

**Comment by borchero** [comment]
@janjagusch I opted for an `--insecure` flag rather than `--verify` to be consistent with `mamba`.

File: src/quetz_client/cli.py [hunk 1]
```diff
@@ -18,10 +21,14 @@ def get_client(*, url=None, token=None) -> QuetzClient:
     token: Optional[str]
         The API key needed to authenticate with the server.
         Defaults to the `QUETZ_API_KEY` environment variable.
+
+    insecure: bool
+        Allow quetz-client to perform "insecure" SSL connections
```
Comment (janjagusch) [inline src/quetz_client/cli.py]:
```suggestion
        Allow quetz-client to perform "insecure" SSL connections.
```

File: src/quetz_client/cli.py [hunk 2]
```diff
@@ -18,10 +21,14 @@ def get_client(*, url=None, token=None) -> QuetzClient:
...
+        and transfers.
     """
     url = url or os.environ.get("QUETZ_SERVER_URL")
     token = token or os.environ.get("QUETZ_API_KEY")
-    client = QuetzClient.from_token(url, token)
+    client = QuetzClient.from_token(url, token, insecure=insecure)
```
Comment (janjagusch) [inline src/quetz_client/cli.py]: Let's not change the client constructor, only the CLI tool.

```suggestion
    client = QuetzClient.from_token(url, token)
    client.session.verify = not insecure
```

**Review by janjagusch — DISMISSED** [review]
Thank you!

**Comment by borchero** [comment]
Sorry for the roundtrip @janjagusch, needed to fix the PCH 😬

**Review by janjagusch — APPROVED** [review]

**Pull request merged**